core/runtime/src/main/java/io/quarkus/runtime/ExecutionMode.java

```diff
@@ -0,0 +1,30 @@
+package io.quarkus.runtime;
+
+/**
+ * The runtime execution mode.
+ */
+public enum ExecutionMode {
+
+    /**
+     * Static initializiation.
+     */
+    STATIC_INIT,
+
+    /**
+     * Runtime initialization.
+     */
+    RUNTIME_INIT,
+
+    /**
+     * The application is running.
+     */
+    RUNNING,
+
+    UNSET,
```

Comment:
IIUC, this could just be called `SHUTDOWN` (or `SHUTTING_DOWN`), no?

Reply:
Right now, we don't care about shutdown. The `UNSET` value is set before the static init phase is started and after the app is shut down.

Reply:
Ah, I looked wrong then. I thought it's only set before shutdown. In that case, `UNSET` is good.